lib/handshake_server.ml

```diff
@@ -241,8 +245,9 @@ let answer_client_hello_reneg state (ch : client_hello) raw =
 
   let process_client_hello config oldsession oldversion ours ch =
     let cciphers = ch.ciphersuites in
+    let fallback = List.mem Packet.TLS_FALLBACK_SCSV cciphers in
     assure (client_hello_valid ch) >>= fun () ->
-    agreed_version config.protocol_versions ch.version >>= fun version ->
+    agreed_version config.protocol_versions ch.version fallback >>= fun version ->
```

Comment:
I kept that line, but I don't think it's useful. The line that checks that version = oldversion is sufficient.